src/pack.c

```diff
@@ -1175,6 +1175,7 @@ int git_packfile_alloc(struct git_pack_file **pack_out, const char *path)
 static git_off_t nth_packed_object_offset(const struct git_pack_file *p, uint32_t n)
 {
 	const unsigned char *index = p->index_map.data;
+	const unsigned char *end = p->index_map.data + p->index_map.len;
```

Comment:
Use `index + p->index_map.len` here or cast the first addend to `const unsigned char *`.